### PR TITLE
Feature/add apartment and floor details to address

### PR DESCRIPTION
**Pull request by AlexSJ1990**
# Jira Issue: [] 

## What?
- 

## Why?
- 

## Optional checklist
- [ ] Codescouted
- [ ] Unit tests written
- [ ] Tested locally

### CODE REVIEW

File: src/main/kotlin/com/hedvig/underwriter/serviceIntegration/memberService/dtos/FinalizeOnBoardingRequest.kt [hunk 1]
```diff
@@ -26,15 +28,16 @@ class FinalizeOnBoardingRequest(
...
-                    zipCode = it.zipCode!!,
-                    apartmentNo = "",
-                    floor = 0
-                )
-            }
+                    Address(
+                        street = addressInsurance.street!!,
+                        city = addressInsurance.city ?: "",
+                        zipCode = addressInsurance.zipCode!!,
+                        apartmentNo = if (addressInsurance is DanishHomeContentAddressData) addressInsurance.apartment ?: "" else "",
```
Comment (Michael-Duivestein) [inline src/main/kotlin/com/hedvig/underwriter/serviceIntegration/memberService/dtos/FinalizeOnBoardingRequest.kt:37]:
```suggestion
                        apartmentNo = (addressInsurance as? DanishHomeContentAddressData)?.apartment ?: "",
```

File: src/main/kotlin/com/hedvig/underwriter/serviceIntegration/memberService/dtos/FinalizeOnBoardingRequest.kt [hunk 2]
```diff
@@ -47,5 +50,15 @@ class FinalizeOnBoardingRequest(
...
+
+        private fun getFloorAsIntegerFromString(floor: String?): Int {
+            if (floor == null) return 0
+
+            return try {
+                floor.toInt()
+            } catch (exception: NumberFormatException) {
+                0
+            }
+        }
```
Comment (Michael-Duivestein) [inline src/main/kotlin/com/hedvig/underwriter/serviceIntegration/memberService/dtos/FinalizeOnBoardingRequest.kt:62]: Could be replaced by `num?.toIntOrNull() ?: 0`

**Review by Michael-Duivestein — APPROVED** [review]

File: src/main/kotlin/com/hedvig/underwriter/serviceIntegration/memberService/dtos/FinalizeOnBoardingRequest.kt [hunk 3]
```diff
@@ -34,8 +33,8 @@ class FinalizeOnBoardingRequest(
                         street = addressInsurance.street!!,
                         city = addressInsurance.city ?: "",
                         zipCode = addressInsurance.zipCode!!,
-                        apartmentNo = if (addressInsurance is DanishHomeContentAddressData) addressInsurance.apartment ?: "" else "",
-                        floor = if (addressInsurance is DanishHomeContentAddressData) getFloorAsIntegerFromString(addressInsurance.floor) else 0
+                        apartmentNo = (addressInsurance as? DanishHomeContentAddressData)?.apartment ?: "",
+                        floor = (addressInsurance as? DanishHomeContentAddressData)?.floor?.toIntOrNull() ?: 0
```
Comment (Michael-Duivestein) [inline src/main/kotlin/com/hedvig/underwriter/serviceIntegration/memberService/dtos/FinalizeOnBoardingRequest.kt:37]: Nice! :D

**Review by Michael-Duivestein — APPROVED** [review]

File: src/main/kotlin/com/hedvig/underwriter/model/Quote.kt [hunk 4]
```diff
@@ -446,12 +446,19 @@ data class Quote(
                     else -> throw IllegalTypeChangeOnQuote(newQuote.data, requestData)
                 }
 
+                val addressInfoHasNotChanged =
+                    requestData.street == null &&
+                    requestData.zipCode == null &&
+                    requestData.apartment == null &&
+                    requestData.floor == null
```
Comment (fredriklagerblad) [inline src/main/kotlin/com/hedvig/underwriter/model/Quote.kt:453]: I notice  `city` has been left out here (and in the copy() method below). Is that intentional ?

Comment (AlexSJ1990) [reply src/main/kotlin/com/hedvig/underwriter/model/Quote.kt:453]: nope good catch!

**Pull request merged**